### ISSUE TITLE
unhelpful test suite error when CH_TEST_*DIR does not exist

### ISSUE THREAD

**Issue by j-ogas** [hpc/charliecloud#347]
If you have `CH_TEST_*DIR` arguments set but the directories do not exist the test suite will give this unhelpful error:

```
$ make test-build
bats build.bats build_auto.bats build_post.bats
/home/jogas/git/charliecloud/test/bats/libexec/bats-exec-suite: line 20: let: count+=: syntax error: operand expected (error token is "+=")
make: *** [test-build] Error 1
```

If the directories do not exist the error message should reflect that. Alternatively, the `CH_TEST_*DIR` directories should just be created if they do not exist and error if they cannot be created at the specified path. FWIW, I much prefer the latter.

**Comment by rstyd** [comment]
Me and Jordan were discussing this earlier, but it seems like some sort of harness around the whole thing to catch these bugs would be helpful.

**Comment by reidpr** [comment]
Something else is going on here, because I get different errors from the above if `CH_TEST_TARDIR` or `CH_TEST_IMGDIR` is set to `/doesnotexist`.

**Comment by reidpr** [comment]
Also, we are already creating both `CH_TEST_IMGDIR` and `CH_TEST_TARDIR` inside the test suite if they don't exist, the former in a somewhat complex way, which complicates my claim above that we shouldn't create them automatically.

**Comment by j-ogas** [comment]
Cannot reproduce. Closing.

**Comment by reidpr** [comment]
Reopening. This seems to happen when the *parent* of specified directory doesn't exist, e.g. if `CH_TEST_IMGDIR=/foo/bar/baz` and `bar` doesn't exist but `foo` does.

**Comment by j-ogas** [comment]
This could possible be addressed with #447.

**Comment by j-ogas** [comment]
I believe this is addressed fixed with the new test feature from #447. Closing.
